### PR TITLE
Recipe App PR

### PR DESCRIPTION
**Pull request by mertkantaroglu**
Hello,

In this project me and my partner @jonas-45;

- Followed the layout of the wireframes provided and created our own design.
- Created Login page and registration page:
    - It was built with Devise.
- Created Food list:
    - It displays a list of food added by the logged-in user as in the wireframe (displays also quantity of a given food).
    - It leads to a form that allows users to add new food.
- Created Recipes list:
    - It displays a list of recipes created by the logged-in user as in the wireframe.
    - It leads to recipe details.
    - If the user is the owner of the recipe, it allows the user to delete it.
- Created Public recipe list:
    - It displays a list of all public recipes ordered by newest as in the wireframe.
    - It leads to recipe details.
- Created Recipe details:
    - It displays a toggle button that allows for a recipe to be made public or private.
    - If the recipe is public or the user is the owner of the recipe, it displays the recipe details as in the wireframe.
    - If the user is the owner of the recipe, it leads to the form that allows the user to add new food.
- Created General shopping list view:
    - It shows the list of food that is missing for all recipes of the logged-in user
    - It counts the total food items and total price of the missing food.
- Fixed N+1 queries problem.
- Created a navigation menu that allows users to open all of the pages.
- Wrote unit and integration tests
- Fixed linter errors

Looking forward to hearing from you.

Thanks.

### CODE REVIEW

**Review by peterrobert — CHANGES_REQUESTED** [review]
Hi @mertkantaroglu  and @jonas-45 , 

Very nice project. Great work on this project. You are almost done👏🏿 . However, there are a few changes to be made kindly check below ⬇️

### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️
[ ] Great work on your design. It looks great ✔️
[ ] Great work on your unit and integration tests. It shows you really took your time to fully test your application.  ✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/recipes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+  # DELETE /recipes/1 or /recipes/1.json
+  def destroy
+    @recipe.destroy
+
+    respond_to do |format|
+      format.html { redirect_to recipes_url, notice: 'Recipe successfully removed' }
+      format.json { head :no_content }
+    end
+  end
```
[the remark this answers is not in the document]
Comment (jonas-45) [reply app/controllers/recipes_controller.rb:58]: Okay, resolving it right away

**Review by Danieldotcomcoder — APPROVED** [review]
Hello @mertkantaroglu , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 👍 👍 😄 👏 

Congratulations! 🎉 🎉 🎉 

## Highlights
- Professional Readme File ✔✔
- Linters are setup correctly ✔✔
- Clear and nice code ✔✔

comp-2
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mertkantaroglu** [comment]
Thank you @Danieldotcomcoder 👍

**Pull request merged**